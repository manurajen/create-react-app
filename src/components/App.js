import React from 'react';

class App extends React.Component{
  render(){
    return(
      <div>
        <h2>Test Driven Developments</h2>
      </div>
    )
  }
}

export default App;