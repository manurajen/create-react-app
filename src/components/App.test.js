import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('renders correctly', () => {
  const component = renderer.create(<App page="http://www.facebook.com">Facebook</App>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders without link', () => {
  const component = renderer.create(<App>Facebook</App>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('class changed when hover over the link', () => {
  const component = renderer.create(<App page="http://www.facebook.com">Facebook</App>,);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

   // manually trigger the callback
  tree.props.onMouseEnter();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

   // manually trigger the callback
  tree.props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});

