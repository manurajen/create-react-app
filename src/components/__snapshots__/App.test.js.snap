// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class changed when hover over the link 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`class changed when hover over the link 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`class changed when hover over the link 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`renders correctly 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`renders without link 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
